src/posix/CMakeLists.txt

```diff
@@ -38,15 +38,32 @@ set(COMMON_INCLUDES
     ${PROJECT_SOURCE_DIR}/src/posix/platform/include
 )
 
-set(OT_READLINE "readline" CACHE STRING "set readline library name")
-set_property(CACHE OT_READLINE PROPERTY STRINGS "readline" "edit")
+set(OT_READLINE_VALUES
+    "readline"
+    "edit"
+)
+
+set(OT_READLINE "" CACHE STRING "set readline library name")
+set_property(CACHE OT_READLINE PROPERTY STRINGS ${OT_READLINE_VALUES})
```

Comment:
What value do we set `OT_READLINE` to disable readline?

Reply:
`OFF/NO/FALSE/0`, see: https://cmake.org/cmake/help/latest/command/if.html#basic-expressions